pkg/visualizer2/connection_graph.go

```diff
@@ -0,0 +1,200 @@
+package visualizer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dominikbraun/graph"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+	"github.com/klothoplatform/klotho/pkg/set"
+)
+
+// ConnectionGraph facilitates whether resources should be connected for visualization. Resources are only considered
+// connected if there are paths that satisfy both "network" and "permissions" classifications.
+type ConnectionGraph struct {
+	Network, Permissions ClassGraph
+	Undirected           construct.Graph
+	KB                   knowledgebase.TemplateKB
+}
+
+func NewConnectionGraph(g construct.Graph, kb knowledgebase.TemplateKB) (ConnectionGraph, error) {
```

Comment:
im confused on what we need this for and where this is managed/ why are network and permissions always required?

Reply:
right but not all connections must satisfy both network and permissions, plus we had the reusable logic for does it have a path so are we not using that anymore? i guess im confused when do we add things to this graph

Reply:
Hm, yeah... This was much different when I started (was the AllPathsBetween version) and when I rebased I didn't notice it had changed. The `path_selection` change was done when I had already moved onto the UI stuff. I'll fix the rebase and probably remove the ConnectionGraph stuff.